builder/buildid.go

```diff
@@ -40,7 +41,16 @@ func ReadBuildID() ([]byte, error) {
 			if uint64(n) != section.Size || err != nil {
 				return nil, fmt.Errorf("could not read build id: %w", err)
 			}
-			return buf, nil
+			if section.Name == ".note.gnu.build-id" {
+				gnuID = buf
```

Comment:
To be squeaky minimal, I'd just return immediately rather than keeping gnuID around...